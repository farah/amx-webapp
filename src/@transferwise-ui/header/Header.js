import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';

const Header = ({ leftContent, rightContent, bottomContent, className }) => (
  <div className={classNames(className)}>
    <div className="container">
      <div className="row p-t-3 ">
        <div className="col-lg-2 col-xs-6">{leftContent}</div>
        <div className="col-lg-2 col-xs-6 col-lg-push-8 text-xs-right">{rightContent}</div>
        <div className="col-xs-12 col-lg-6 col-lg-pull-2 col-lg-offset-1 p-x-0">
          {bottomContent}
        </div>
      </div>
    </div>
  </div>
);

Header.defaultProps = {
  leftContent: null,
  rightContent: null,
  bottomContent: null,
  className: null,
};

Header.propTypes = {
  leftContent: Types.node,
  rightContent: Types.node,
  bottomContent: Types.node,
  className: Types.string,
};

export default Header;
