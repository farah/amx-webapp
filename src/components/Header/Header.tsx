import React from "react";
import { NavLink, Link } from "react-router-dom";
import Logo from "components/Logo";
import { useAuth } from "contexts/Firebase";
import styled from "styled-components";

const StyledLink = styled(NavLink)`
  color: ${props => props.color};
`;

interface Props {
  color?: any;
}
const Header: React.FC<Props> = ({ color }) => {
  const [open, setOpen] = React.useState(false);

  const toggleMobile = () => {
    setOpen(!open);
  };

  const closeMobile = () => {
    setOpen(false);
  };

  const { authUser } = useAuth();

  return (
    <header className="globalNav initialized noDropdownTransition">
      <div className="container-lg">
        <nav>
          <ul onBlur={toggleMobile} className="navRoot">
            <li className="navSection logo">
              <Logo />
            </li>

            <li className="navSection secondary">
              {authUser ? (
                <>
                 <StyledLink className="rootLink item-support" color={color} to="/contact">
                  <span>Support</span>
                </StyledLink>
                <StyledLink className="rootLink item-support" color={color} to="/transfer">
                  <span>Dashboard</span>
                </StyledLink>

                </>
              ) : (
                <>
                <StyledLink className="rootLink item-support" color={color} to="/contact">
                  <span>Support</span>
                </StyledLink>
                <StyledLink className="rootLink item-support" color={color} to="/login">
                  <span>Login</span>
                </StyledLink>
                </>
              )}
            </li>
            <li
              onBlur={closeMobile}
              className={`navSection mobile ${open ? "globalPopupActive" : ""}`}
            >
              <a
                className={`rootLink item-mobileMenu colorize`}
                onBlur={toggleMobile}
                onClick={toggleMobile}
              >
                <h2>Menu</h2>
              </a>
              <div className="popup">
                <div className="popupContainer">
                  <a onClick={toggleMobile} className="popupCloseButton">
                    Close
                  </a>
                  <div className="mobileProducts">
                    <h4>Pages</h4>
                  </div>
                  <div className="mobileSecondaryNav">
                    <ul>
                      <li>
                        <Link className="linkContainer item-enterprise" to="/locations">
                          Locations
                        </Link>
                      </li>
                      <li>
                        {authUser ? (
                          <Link className="item-workswith" to="/transfer">
                            Dashboard
                          </Link>
                        ) : (
                          <Link className="item-workswith" to="/contact">
                            Support
                          </Link>
                        )}
                      </li>
                    </ul>
                  </div>

                  <Link className="mobileSignIn" to="/login">
                  Sign in
                          </Link>
                </div>
              </div>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  );
};

export default Header;
