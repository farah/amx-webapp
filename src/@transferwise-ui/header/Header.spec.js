import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  it('renders correctly', () => {
    const props = {
      leftContent: 'leftContent',
      rightContent: 'rightContent',
      bottomContent: 'bottomContent',
      className: 'className',
    };
    const component = shallow(<Header {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });
});
