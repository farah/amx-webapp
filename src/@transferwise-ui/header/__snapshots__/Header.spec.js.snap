// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
"<div className=\\"className\\">
  <div className=\\"container\\">
    <div className=\\"row p-t-3 \\">
      <div className=\\"col-lg-2 col-xs-6\\">
        leftContent
      </div>
      <div className=\\"col-lg-2 col-xs-6 col-lg-push-8 text-xs-right\\">
        rightContent
      </div>
      <div className=\\"col-xs-12 col-lg-6 col-lg-pull-2 col-lg-offset-1 p-x-0\\">
        bottomContent
      </div>
    </div>
  </div>
</div>"
`;
